llarp/util/time.hpp

```diff
@@ -29,7 +25,64 @@ namespace llarp
   struct time_delta
   {
     const TimePoint_t at;
+
+    /// how far away are we from the time point we have
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    delta() const
+    {
+      return distance_from<delta_res_t>(DateClock_t::now());
+    }
+
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    distance_from(const TimePoint_t& other) const
+    {
+      return std::chrono::duration_cast<delta_res_t>(other - at);
+    }
   };
+
+  /// get a time delta with a timepoint in the future
+  template <typename Time_Duration>
+  TimePoint_t
+  future_time_point(Time_Duration future)
+  {
+    return DateClock_t::now() + future;
+  }
```

Comment:
Is this function really needed?  Why not just write `DateClock_t::now() + x` where you want this?

Reply:
no it is not needed.